core/src/autogluon/core/searcher/bayesopt/tuning_algorithms/base_classes.py

```diff
@@ -116,6 +143,10 @@ def backward_gradient(
         pass
 
 
+# Useful type that allows for a dictionary mapping each output name to a SurrogateModel
```

Comment:
Can you expand this docstring? This type def is used all over the place. Please do explain why this is sometimes a dict.